Core/MigrationService.php

```diff
@@ -437,7 +440,22 @@ protected function executeMigrationInner(Migration $migration, MigrationDefiniti
                 $currentUser = $this->getCurrentUser();
                 $this->authenticateUserByLogin($adminLogin ?? self::ADMIN_USER_LOGIN);
 
-                $this->repository->commit();
+                try {
+                    $this->repository->commit();
+                } catch (Exception $e) {
+                    if ($executor && !$executor instanceof SQLExecutor) {
```

Comment:
This check is not always correct, as we can have the sqlExecutor kick in f.e. in migration step 1, and another executor in step 2 (it is possible to have sql steps in yaml migrations...)